### ISSUE TITLE
Compare AIX openjdk jtreg exclusion list with SAP's

### ISSUE THREAD

**Issue by sxa** [adoptium/aqa-tests#2828]
We should compare SAP's list of AIX test exclusions [as posted to the ppc-aix-port-dev list](https://mail.openjdk.java.net/pipermail/ppc-aix-port-dev/2021-July/thread.html) with the [exclusions that we have in this repository](https://github.com/adoptium/aqa-tests/tree/master/openjdk/excludes), and sync them as appropriate to ensure consistent coverage with the team that had been working on the port as it transitions to another custodian.

**Comment by BrijeshNekkare** [comment]
We observe a large amount of differences between SAP JDK-8 tests excluded in

> 	https://mail.openjdk.java.net/pipermail/ppc-aix-port-dev/2021-July/004127.html
> 	https://mail.openjdk.java.net/pipermail/ppc-aix-port-dev/2021-July/004128.html
> 

Versus the one maintained at Adoptium repo
> 	https://github.com/adoptium/aqa-tests/blob/master/openjdk/excludes/ProblemList_openjdk8.txt

Minimal set of AIX specific test exclusions (tag=aix-all) can be seen at OpenJDK repo: https://github.com/openjdk/jdk8u/blob/master/jdk/test/ProblemList.txt and another few on top of it with tag=generic-all

So,
1. Are the SAP test excludes listed in email, maintained internally by them in a non-public repo
2. Are there other excludes maintained in Adoptium apart from the one listed [here](https://github.com/adoptium/aqa-tests/blob/master/openjdk/excludes/ProblemList_openjdk8.txt)

**Comment by BrijeshNekkare** [comment]
Observe Test groups are listed at [here](https://github.com/adoptium/jdk8u/blob/c7691288e1d7a9dd52e589967716fb425057fcd7/jdk/test/TEST.groups)

**For example - jdk_security3**

> jdk_security3 = \
>     javax/security  \
>     -javax/security/auth/kerberos \
>     com/sun/security \
>     -com/sun/security/jgss \
>     com/sun/org/apache/xml/internal/security \
>     sun/security \
>     -sun/security/krb5 \
>     -sun/security/jgss \
>     javax/net \
>     lib/security
> 

I read the above as : 
All test folders listed under **jdk_security3** group should be executed, exception being the ones preceded by '-' as in -javax/security/auth/kerberos and other similar such entries. Is my understanding correct?

To Summarize, here is how i should proceed:
1. Pick the excluded test case mentioned by SAP 
2. Check the directory under which it exists and its relevant Test group
3. See if it belongs to the disabled list of Test group in [playlist.xml](https://github.com/adoptium/aqa-tests/blob/master/openjdk/playlist.xml) 

Or is there is a better faster way to accomplish the comparison task?

**Comment by BrijeshNekkare** [comment]
Thanks to @ShelleyLambert,  Am able to rule out some more tests based on targets disabled in [playlist.xml](https://github.com/adoptium/aqa-tests/blob/master/openjdk/playlist.xml) file

jdk_swing are currently disabled for all platforms including AIX.

> Test group: 
> 	jdk_swing = javax/accessibility javax/swing com/sun/java/swing
> Tests:
> 	javax/swing/JEditorPane/8195095/ImageViewTest.java                                       generic-all
> 	javax/swing/JProgressBar/8161664/ProgressBarMemoryLeakTest.java              generic-all
> 

jdk_awt are currently disabled for all platforms including AIX.

> Test group: 
> 	jdk_awt = java/awt com/sun/awt com/apple/eawt sun/awt
> Tests:	
> 	java/awt/Container/MoveToOtherScreenTest/MoveToOtherScreenTest.java                  generic-all
> 	java/awt/event/SequencedEvent/MultipleContextsFunctionalTest.java                         generic-all
> 	java/awt/Focus/NullActiveWindowOnFocusLost/NullActiveWindowOnFocusLost.java  generic-all
> 	java/awt/Mouse/EnterExitEvents/ModalDialogEnterExitEventsTest.java                         generic-all
> 	java/awt/ScrollPane/ScrollPaneValidateTest.java                                                             generic-all
> 	java/awt/datatransfer/SystemFlavorMap/AddFlavorTest.java
> 	java/awt/font/LineBreakMeasurer/FRCTest.java                                 generic-ppc64
> 	java/awt/Toolkit/BadDisplayTest/BadDisplayTest.java                          aix-all
> 	java/awt/FontClass/CreateFont/fileaccess/FontFile.java                       aix-all
> 	java/awt/JAWT/JAWT.sh                                                        aix-all
> 	java/awt/Toolkit/AutoShutdown/EventQueuePush/EventQueuePushAutoshutdown.sh   aix-all
> 	java/awt/Graphics2D/DrawString/RotTransText.java                             aix-all
> 	sun/awt/datatransfer/SuplementaryCharactersTransferTest.java
> 
> 

jdk_2d are currently disabled for all platforms including AIX.

> Test group: 
> 	jdk_2d =     javax/print     sun/pisces      sun/java2d
> Tests:	
> 	sun/java2d/cmm/ColorConvertOp/ColConvCCMTest.java                            aix-all
> 	sun/java2d/cmm/ColorConvertOp/MTColConvTest.java                               aix-all

jdk_security3 are currently disabled for all platforms including AIX

> Test group: 
> 	jdk_security3 =     javax/security      -javax/security/auth/kerberos     com/sun/security     
> 	-com/sun/security/jgss com/sun/org/apache/xml/internal/security     sun/security     -sun/security/krb5     
> 	-sun/security/jgss     javax/net   lib/security
> Tests:
> 	sun/security/tools/jarsigner/diffend.sh                                      aix-all

Still a fair way to go for completion

**Comment by BrijeshNekkare** [comment]
Following targets are enabled in playlist.xml file, whereas SAP has excluded the specific tests

> Test group: 
>      jdk_other = \
>     java/sql     javax/sql     javax/rmi     javax/naming     javax/script     javax/smartcardio     javax/xml 
>     -javax/xml/crypto     jdk/asm     com/sun/jndi     com/sun/corba     lib/testlibrary     demo/zipfs     sample
> Tests :
> 	javax/xml/ws/publish/WSTest.java                                             generic-all
> 

> Test group: 
>         jdk_security_infra =     security/infra/java/security/cert/CertPathValidator/certification
> Tests:	
> security/infra/java/security/cert/CertPathValidator/certification/BuypassCA.java      generic-all
> 	security/infra/java/security/cert/CertPathValidator/certification/ComodoCA.java       generic-all
> 	security/infra/java/security/cert/CertPathValidator/certification/LetsEncryptCA.java  generic-all
> 	security/infra/java/security/cert/CertPathValidator/certification/QuoVadisCA.java     generic-all
> 
> 

> Test group: 
> jdk_beans =     java/beans
> Tests:
> 	java/beans/Introspector/7064279/Test7064279.java              windows-all,solaris-all
> 	java/beans/Introspector/Test7172865.java                      windows-all,solaris-all
> 	java/beans/Introspector/Test7195106.java                      windows-all,solaris-all
> 	java/beans/Introspector/Test8027905.java                      windows-all,solaris-all
> 
> 

> Test group:
>  jdk_net =     java/net     com/sun/net     com/oracle/net     sun/net     jdk/net
> Tests:
> 	java/net/MulticastSocket/TestInterfaces.java                                            solaris-all
> 	java/net/NetworkInterface/UniqueMacAddressesTest.java                                   solaris-all
> 	java/net/URLConnection/HandleContentTypeWithAttrs.java        windows-all,generic-ppc64,solaris-all
> 	java/net/ServerSocket/AcceptInheritHandle.java                               aix-all
> 	java/net/Socket/InheritHandle.java                                           aix-all
> 
> 

> Test group: 
> jdk_nio =     java/nio     sun/nio
> Tests:
> 	java/nio/file/Files/CopyAndMove.java                          windows-all
> 	java/nio/file/Files/delete_on_close.sh                                       aix-all
> 	java/nio/file/Files/SBC.java                                                 aix-all
> 
> 

> Test group: 
> jdk_imageio =     javax/imageio
> Tests:
> 	javax/imageio/plugins/bmp/BMP8BPPLoadTest.java                windows-all,generic-ppc64,solaris-all
> 
> 

> Test group: 
> jdk_security4 =     com/sun/security/jgss     javax/security/auth/kerberos     sun/security/krb5     sun/security/jgss
> Tests:
> 	sun/security/krb5/auto/Unreachable.java                                      aix-all
>

**Comment by BrijeshNekkare** [comment]
**Pending tests to check:**

jdk/security/infra/java/security/cert/CertPathValidator/certification/HaricaCA.java   generic-all
sun/security/tools/keytool/printssl.sh                        windows-all

runtime/CompressedOops/CompressedClassPointers.java         generic-ppc64
runtime/NMT/VirtualAllocCommitUncommitRecommit.java         generic-ppc64

compiler/ciReplay/TestSA.sh                             generic-all
compiler/ciReplay/TestVM_no_comp_level.sh               generic-all
compiler/ciReplay/TestVM.sh                             generic-all

runtime/SharedArchiveFile/SharedBaseAddress.java            aix-ppc64

serviceability/dcmd/DynLibDcmdTest.java                        aix-ppc64

gc/g1/TestLargePageUseForAuxMemory.java                 generic-all

compiler/6478991/NullCheckTest.java              generic_ppc64
compiler/6579789/Test6579789.java                generic_ppc64
compiler/6756768/Test6756768_2.java              generic_ppc64
compiler/6756768/Test6756768.java                generic_ppc64
compiler/6757316/Test6757316.java                generic_ppc64
compiler/6758234/Test6758234.java                generic_ppc64
compiler/6769124/TestArrayCopy6769124.java       generic_ppc64
compiler/6769124/TestDeoptInt6769124.java        generic_ppc64
compiler/6769124/TestUnalignedLoad6769124.java   generic_ppc64
compiler/6795465/Test6795465.java                generic_ppc64
compiler/6849574/Test.java                       generic_ppc64
compiler/6855215/Test6855215.java                generic_ppc64
compiler/6932496/Test6932496.java                generic_ppc64
compiler/7042153/Test7042153.java                generic_ppc64
compiler/7090976/Test7090976.java                generic_ppc64
compiler/7103261/Test7103261.java                generic_ppc64
compiler/7123108/Test7123108.java                generic_ppc64
compiler/8011706/Test8011706.java                generic_ppc64
compiler/8011771/Test8011771.java                generic_ppc64

compiler/exceptions/TestRecursiveReplacedException.java generic-ppc64
compiler/runtime/6865265/StackOverflowBug.java          linux-ppc64
compiler/6865265/StackOverflowBug.java                  linux-ppc64

compiler/uncommontrap/8009761/Test8009761.java          linux-ppc64,aix-ppc64,windows-all
compiler/8009761/Test8009761.java                       linux-ppc64,aix-ppc64,windows-all
compiler/uncommontrap/StackOverflowGuardPagesOff.java   linux-ppc64
compiler/uncommontrap/TestStackBangMonitorOwned.java    linux-ppc64
compiler/uncommontrap/TestStackBangRbp.java             linux-ppc64

gc/arguments/TestMaxHeapSizeTools.java                  linux-ppc64

runtime/whitebox/WBStackSize.java                       linux-ppc64

sanity/ExecuteInternalVMTests.java                      linux-ppc64

gc/arguments/TestCMSHeapSizeFlags.java                  linux-ppc64
gc/arguments/TestG1HeapSizeFlags.java                   linux-ppc64
gc/arguments/TestParallelHeapSizeFlags.java             linux-ppc64
gc/arguments/TestSerialHeapSizeFlags.java               linux-ppc64

testlibrary_tests/TestMutuallyExclusivePlatformPredicates.java aix-ppc64

gc/metaspace/CompressedClassSpaceSizeInJmapHeap.java    generic-all,macosx-all
gc/metaspace/CompressedClassSpaceSizeInJmapHeap.jtr     generic-all,macosx-all

compiler/arguments/CheckCompileThresholdScaling.java    generic-ppc64
compiler/6982370/Test6982370.java                       generic-ppc64

runtime/CompressedOops/UseCompressedOops.java           aix-ppc64

**Comment by BrijeshNekkare** [comment]
Am unsure about how to confirm the pending test case exclusions.

If I have access to Jenkins console output when the full suite of test cases was run for AIX platform, perhaps that is one way. If this is the right approach can someone please point me to the Jenkins link. Any help most appreciated.

**Comment by smlambert** [comment]
Anyone should be able to see the console output of the test runs.

Many of the test targets being discussed in this issue are categorized into the extended.openjdk group, which can be found here for jdk8:
https://ci.adoptopenjdk.net/view/Test_openjdk/job/Test_openjdk8_hs_extended.openjdk_ppc64_aix/ 

So looking at a recent [job 25](https://ci.adoptopenjdk.net/view/Test_openjdk/job/Test_openjdk8_hs_extended.openjdk_ppc64_aix/25/), we run it in parallel across 3 machines, you can see the 3 child test jobs and click on each to see their console output, and look at [TAP Extended Test Results](https://ci.adoptopenjdk.net/view/Test_openjdk/job/Test_openjdk8_hs_extended.openjdk_ppc64_aix/25/tapResults/) to see which test targets were included in them, while each of the child jobs [Test Result link](https://ci.adoptopenjdk.net/view/Test_openjdk/job/Test_openjdk8_hs_extended.openjdk_ppc64_aix_testList_0/8/testReport/) shows the Junit output of the individual test cases (and you can look at each child's console output also).

**Comment by smlambert** [comment]
Closing this issue as "as done as its going to get".
